### ISSUE TITLE
Updating threads in multiple channels doesn't work correctly

### ISSUE THREAD

**Issue by dbasilio** [slackapi/slack-github-action#229]
### Description

#118 introduced the ability to post a message to multiple channels, but we're having issues with updating the threads in multiple channels afterwards. When calling the action you can put multiple channel IDs and and those get posted to correctly. However, the output from the action does not include the channel and thread id for each channel, only the last one to respond.

I'm happy to fix this bug, but want to confirm what the resulting output should be since the change is likely going to be breaking.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [x] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

The issue stems from [here](https://github.com/slackapi/slack-github-action/blob/10d6bcdbd1f8a758718aa384c8ec40281e75aa13/src/slack-send.js#L77-L85) it's always overwriting the `webResponse` variable, so then when you get to [the output section](https://github.com/slackapi/slack-github-action/blob/10d6bcdbd1f8a758718aa384c8ec40281e75aa13/src/slack-send.js#L145-L152) it only holds a reference to a single channel and thread id instead of all of them.

In order to fix this, you may need to change the shape of the output (which would be a breaking change). I'd suggest the following structure:
```json
{
    "threads": [
        { "ts": "...", "thread_ts": "...", "channel_id": "..." },
        { "ts": "...", "thread_ts": "...", "channel_id": "..." }
    ]
}
```
An alternative would be to return a comma-separate list in each of ts, thread_ts, and channel_id, but I think it's safer to do an array to make sure the right thread_ts and channel_id are kept together.

#### Steps to reproduce:

1. Call the action with multiple channels
2. Store the output from the action in a workflow for use later
3. View the output and see that only a single channel was output.

#### Expected result:

The output from the action includes the channel and thread ids for all the channels messaged.

#### Actual result:

The output from the action includes whichever channel/thread was the last to resolve from the API.

#### Attachments:

Logs showing we called the action with multiple channel IDs
![image](https://github.com/slackapi/slack-github-action/assets/8311284/1ca4e88a-f438-4710-869a-d37fd7c7547f)

Log showing the output from the output (stored in an env_var METADATA_JSON)
![image](https://github.com/slackapi/slack-github-action/assets/8311284/d3ea9d4a-c2f5-4002-bede-c9b23d7fe666)

**Comment by dbasilio** [comment]
Actually as I look at the code more to properly support updating multiple channels the inputs would likely need to change. Right now `input-ts` is not split, so you can only input a single `input-ts` value, but if you are trying to update threads in multiple channels that would not work :\

**Comment by seratch** [comment]
Hi, thanks for writing in. To my understanding, the multi-channel posting feature was added to simply support posting a single message across multiple channels. I don't believe it was designed to spread the same thread discussion across several channel threads, and that remains true today.

As one of the maintainers of this tool, I hesitate to add such complex functionality to the project at this time. However, if many other users express the same need or if other maintainers feel strongly that we should support this use case, I would be open to enhancing it.

In the meantime, if you need this feature immediately, please consider forking this project or starting your own. I'd appreciate it if you could understand this.

**Comment by dbasilio** [comment]
@seratch Of course! Thanks for the quick reply! It does feel very much like a full breaking change to both inputs and outputs so taking time to consider those effects is well warranted. I think for our use case we're going to move to calling the action multiple times, once for each channel we want to keep up to date. I will keep an eye on the discussion though!

**Comment by PramodKumarYadav** [comment]
Hi @seratch , 

I wanted to use this feature to trigger e2e tests post new deployments and update team on a team channel and individuals using the bot. 

Since e2e tests take some time to finish, I wanted to start by posting a message that tests has started and then later update the same message on both channel and to the individual. 

The reason, I want both of these to be updated is, if we only send replies to individuals, team does not know if a person misses it. If we only post it on channel, then it is expected that most people will not watch it since it gets noisy after a while. 

So the above use case was a perfect solution (if it could work).

**Comment by zimeg** [comment]
@PramodKumarYadav I'm wondering if [a matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) with different channel IDs might serve as a workaround for posting and updating the same message at different points in a workflow? 🤔

I plan to explore this a bit soon since the current progress of `@v2` doesn't support multiple channel IDs in this input to mirror [`chat.postMessage`](https://api.slack.com/methods/chat.postMessage), but this seems like a useful case to support at an action level!

**Comment by PramodKumarYadav** [comment]
Thanks for the suggestion @zimeg. 

In my use case, I am trying to send a message to the github _actor_ whose last commit (and merge to main) triggered the deployment. 

I am dynamically getting this value from github context and storing a map of actor name with their slack-ids in github secrets. This is how, I am passing both the "fixed" channel name and "dynamic" slack id of actor.  

`channel-id: 'C06MYNAS400,${{ secrets[github.event.client_payload.github.actor] }}'`

Since matrix strategy expects hard coded values and I can only know about the triggering person at the time of running tests, unfortunately in this use case, I am not able to solve this using matrix strategy.

If you can think or suggest any workarounds, I will be happy to hear.

**Comment by zimeg** [comment]
@PramodKumarYadav Woah TIL that matrix values must be hardcoded! It makes sense, but that stumped me a bit...

I know this might not be ideal, even a bit repetitive, but would duplicating the `slack-send` step for each of the messages being sent work? This should let these steps use a separate `id`, each with output related to the message that was sent to just the one channel, in following steps? :eyes:

**Comment by zimeg** [comment]
👋 Hello again! We've just released version [`@v2.0.0`](https://github.com/slackapi/slack-github-action/releases/tag/v2.0.0) and I think duplicating the step that sends the `payload` is a decent approach at the moment for keeping track of individual thread timestamps, so will close this issue with a workaround.

The following steps can be used for [the initial message and thread](https://github.com/slackapi/slack-github-action?tab=readme-ov-file#replying-to-a-message), then these could be repasted as much as you'd like with changing IDs:

```yaml
- name: Initiate a deployment
  uses: slackapi/slack-github-action@v2.0.0
  id: deployment_message_1
  with:
    method: chat.postMessage
    token: ${{ secrets.SLACK_BOT_TOKEN }}
    payload: |
      channel: ${{ secrets.SLACK_CHANNEL_ID }}
      text: "Deployment started :eyes:"

# Include duplicates of the shown steps and additional deployment steps here!

- name: Conclude the deployment
  if: ${{ steps.deployment_message_1.outputs.ok }}
  uses: slackapi/slack-github-action@v2.0.0
  with:
    method: chat.postMessage
    token: ${{ secrets.SLACK_BOT_TOKEN }}
    payload: |
      channel: "${{ steps.deployment_message_1.outputs.channel_id }}"
      thread_ts: "${{ steps.deployment_message_1.outputs.ts }}"
      text: "Deployment finished! :rocket:"
```

The channel ID can be a conversation of all kind, and user IDs can be used to send DMs from the bot. More info on collecting a user ID in an action might be found in #349 too 🙏

Please note: Sending a message to multiple channels with one step is no longer supported with this update as well!

Feel free to comment if this workaround isn't working well with some of these changes or if you have other suggetions. Also, opening [a new issue](https://github.com/slackapi/slack-github-action/issues/new) with other questions and ideas is always encouraged! 💡
